src/main/java/ru/all_easy/push/telegram/commands/rules/AutoMigrationRule.java

```diff
@@ -0,0 +1,40 @@
+package ru.all_easy.push.telegram.commands.rules;
+
+import org.springframework.stereotype.Service;
+import ru.all_easy.push.common.ResultK;
+import ru.all_easy.push.currency.service.CurrencyService;
+import ru.all_easy.push.room.service.RoomService;
+import ru.all_easy.push.telegram.api.controller.model.Update;
+import ru.all_easy.push.telegram.commands.rules.model.CommandError;
+import ru.all_easy.push.telegram.commands.rules.model.CommandProcessed;
+
+@Service
+public class AutoMigrationRule implements CommandRule {
+    private final RoomService roomService;
+    private final CurrencyService currencyService;
+
+    public AutoMigrationRule(RoomService roomService, CurrencyService currencyService) {
+        this.roomService = roomService;
+        this.currencyService = currencyService;
+    }
+
+    @Override
+    public boolean apply(Update update) {
+        if (update.message() == null) return false;
+        return update.message().migrateFromChatId() != null;
+    }
+
+    @Override
+    public ResultK<CommandProcessed, CommandError> process(Update update) {
+        final String oldToken = String.valueOf(update.message().migrateFromChatId());
+        final String newToken = String.valueOf(update.message().chat().id());
+
+        try {
+            roomService.autoMigrateToSupergroup(oldToken, newToken);
+        } catch (Exception e) {
+            return ResultK.Err(new CommandError(update.message().chat().id(), e.getMessage()));
```

Comment:
Bad practice to send user exception message, better to create some readable massage like - something went wrong and log exception message with logger

Reply:
Done